### DISCUSSION TITLE
Resizing unix shells with Windows Terminal doesn't clear the previous prompt, spamming up the shell and eventually crashing windows terminal.

### DISCUSSION THREAD

**Discussion by AskAlice** [JanDeDobbeleer/oh-my-posh#403] [Troubleshoot] [answered]
### Prerequisites

- [x] I have read and understand the `CONTRIBUTING` guide
- [x] I looked for duplicate issues before submitting this one

### Description

Resizing unix shells via WSL or Windows OpenSSH doesn't clear the previous prompt, spamming up the shell and eventually crashing windows terminal.

HQ version: https://giant.gfycat.com/MiniatureThinGreatdane.webm

<img src="https://thumbs.gfycat.com/MiniatureThinGreatdane-size_restricted.gif" />

### Environment

- Oh my Posh version: 3.8.6
- Theme: slim
- Operating System: Windows 10 with WSL, and SSH
- Shell: zsh
- Terminal: Windows Terminal

### Steps to Reproduce

1. Open up Windows Terminal
2. Open up a WSL or SSH session that uses zsh and oh-my-posh
3. resize the window for 10 seconds
4. crash windows terminal

**Expected behavior:** no crash, and also it should clear the prompt.

**Actual behavior:** it doesn't clear the prompt, and then crashes the terminal.

**Comment by JanDeDobbeleer** [comment]
@AskAlice I understand the issue, but how is that the responsibility of the prompt generator? Oh my Posh hooks into the shell's prompt function (depends on the platform). Clearing also implies clearing the screen which is maybe not what the user wants (clearing a line is weird as you don't even know where the cursor is at that point in time). Maybe I'm not aware of something but I don't see how we can do this without opening another jar of impossible to resolve issues.

**Comment by JanDeDobbeleer** [comment]
@AskAlice I found [this](https://github.com/microsoft/terminal/issues/8596) issue, which seems pretty close to what you're experiencing. For what it's worth, I tried to reproduce this using iTerm2 on Mac and I can't make the app crash when reproducing this.

**Comment by just1a-person** [comment]
That's right. The terminal isn't supposed to crash because of a prompt. Looking at the issue linked, This seems to be an issue with the headless Console Host that powers the Windows Terminal

**Comment by AskAlice** [comment]
I went to make this issue because of the spamming of the prompt, and found it also happened to crash the terminal when done for long enough

**Comment by AskAlice** [comment]
> @AskAlice I understand the issue, but how is that the responsibility of the prompt generator? Oh my Posh hooks into the shell's prompt function (depends on the platform). Clearing also implies clearing the screen which is maybe not what the user wants (clearing a line is weird as you don't even know where the cursor is at that point in time). Maybe I'm not aware of something but I don't see how we can do this without opening another jar of impossible to resolve issues.

If i use p10k with .oh-my-zsh it doesn't spam like this, so it's somehow possible within this repo to fix that, not quite sure where to start, though. 

They have this [special prompt that doesn't show previous prompts after you hit enter](https://github.com/romkatv/powerlevel10k#transient-prompt), just an arrow then what you wrote in the prompt, then the output, and eventually the current (and only visible) prompt.

**Comment by JanDeDobbeleer** [comment, marked as answer]
@AskAlice I also use zsh and the prompt is't being spammed in resize. Somehow it feels like Windows Terminal is doing some voodoo magic underneath. p10k leverages zsh directly, something we can't do (other than the init script). There is also more information to be found in other [repos](https://github.com/Powerlevel9k/powerlevel9k/issues/881) that talks directly about similar [behaviour](https://github.com/romkatv/powerlevel10k/issues/175) caused by `RPROMPT`. Coming back to the root cause of the issue, namely that your prompt is being replicated (zsh issue) and that the terminal crashes (windows terminal issue) 😅

**Comment by AskAlice** [comment]
Alright feel free to close if that's the case

**Comment by JanDeDobbeleer** [comment]
@AskAlice maybe there's a solution in those issues we can implement in the zsh [setup](https://github.com/JanDeDobbeleer/oh-my-posh3/blob/main/src/init/omp.zsh). I'll have to read through it more closely.

**Comment by JanDeDobbeleer** [comment]
@AskAlice a solution would be to disable wrap on resize in the terminal, however, the Windows Terminal team [decided against](https://github.com/microsoft/terminal/issues/1860) adding the functionality (lord knows why).
